cql3/type_json.cc

```diff
@@ -151,12 +151,16 @@ static bytes from_json_object_aux(const map_type_impl& t, const rjson::value& va
     std::map<bytes, bytes, serialized_compare> raw_map(t.get_keys_type()->as_less_comparator());
     for (auto it = value.MemberBegin(); it != value.MemberEnd(); ++it) {
         bytes value = from_json_object(*t.get_values_type(), it->value);
-        if (!t.get_keys_type()->is_compatible_with(*utf8_type)) {
+        if (!t.get_keys_type()->is_compatible_with(*ascii_type)) {
```

Comment:
probably a more readable way to implement this is

```c++
if (t.get_keys_type()->is_compatible_with(*ascii_type) ||
    t.get_keys_type()->is_compatible_with(*utf8_type)) {
    raw_map.emplace(from_json_object(*t.get_keys_type(), it->name), std::move(value));
} else {
  // reparse 
}
```
for two reasons:

1. we don't encode the knowledge of UTF-8 is compatible with ASCII here.
2. less negative conditions.

Reply:
I agree that it's clearer. 
Another option is to stop using is is_compatible_with and check for actual equality of the types, which I think we can. Or using underlying_type(), in case we happen to be dealing with a reversed clustering key. Something like:
```c++
if (t.get_keys_type()->underlying_type() == ascii_type) ||
    t.get_keys_type()->underlying_type() == utf8_type)) {
    raw_map.emplace(from_json_object(*t.get_keys_type(), it->name), std::move(value));
} else {
  // reparse 
}
```